pyhip/hip.py

```diff
@@ -383,6 +384,231 @@ def hipCheckStatus(status):
+    operations.
+
+    This command honors the hipDeviceLaunchBlocking flag, which controls whether the wait is active
+    or blocking.
+
+    Parameters
+    -------
+    ptr : ctypes pointer
+        Valid pointer to stream object.
+    """
+    status = _libhip.hipStreamSynchronize(ptr)
+    hipCheckStatus(status)
+
+# Event management
+
+# Event creation flags
+hipEventDefault = 0
+hipEventBlockingSync = 1
+hipEventDisableTiming = 2
+hipEventInterprocess = 4
+
+_libhip.hipEventCreateWithFlags.restype = int
+_libhip.hipEventCreateWithFlags.argtypes = [ctypes.POINTER(ctypes.c_void_p), ctypes.c_int]
+def hipEventCreateWithFlags(flags):
+    """
+    Create an event with the specified flags
+
+    Parameters
+    ----------
+    flags : int
+        Flags to control event behavior.  Valid values are hipEventDefault,
+        hipEventBlockingSync, hipEventDisableTiming, #hipEventInterprocess
+        * hipEventDefault : Default flag.  The event will use active synchronization and will support
+        timing.  Blocking synchronization provides lowest possible latency at the expense of dedicating a
+        CPU to poll on the event.
+        * hipEventBlockingSync : The event will use blocking synchronization : if hipEventSynchronize is
+        called on this event, the thread will block until the event completes.  This can increase latency
+        for the synchroniation but can result in lower power and more resources for other CPU threads.
+        * hipEventDisableTiming : Disable recording of timing information. Events created with this flag
+        would not record profiling data and provide best performance if used for synchronization.
+        * hipEventInterprocess : Warning On AMD platform, support is under development.  Use of this flag
+        will return an error.
+
+    Returns
+    -------
+    ptr : ctypes pointer
+        Pointer to the newly created event.
+
+    """
+    ptr = ctypes.c_void_p()
+    status = _libhip.hipEventCreateWithFlags(ctypes.byref(ptr), flags)
+    hipCheckStatus(status)
+    return ptr
+
+_libhip.hipEventCreate.restype = int
+_libhip.hipEventCreate.argtypes = [ctypes.POINTER(ctypes.c_void_p)]
+def hipEventCreate():
+    """
+    Create an event with the specified flags
+
+    Returns
+    -------
+    ptr : ctypes pointer
+        Pointer to the newly created event.
+
+    """
+    ptr = ctypes.c_void_p()
+    status = _libhip.hipEventCreate(ctypes.byref(ptr))
+    hipCheckStatus(status)
+    return ptr
+
+_libhip.hipEventRecord.restype = int
+_libhip.hipEventRecord.argtypes = [ctypes.c_void_p, ctypes.c_void_p]
+def hipEventRecord(event, stream=None):
+    """
+    Record an event in the specified stream.
+
+    Parameters
+    ----------
+    event : handle
+        event to record.
+    stream : _ctypes pointer, optional
+        stream in which to record event.
+
+    """
+    status = _libhip.hipEventRecord(event, stream)
```

Comment:
there needs to be a check.

If stream is None, we need to pass ctypes nullptr instead of None directly to HIP API.